include/megaapi.h

```diff
@@ -5874,6 +5914,77 @@ class MegaListener
+ * completed afterward.
+ *
+ */
+
+class MegaBackgroundMediaUpload
+{
+public:
+
+    /**
+     * @brief Extract mediainfo information about the photo or video.
+     *
+     * Call this function once with the file to be uploaded.  It uses mediainfo to extract information that will
+     * help the webclient show or play the file in various browsers.  The information is stored in this object
+     * until the whole operation completes.
+     *
+     * @param inputFilepath The file to analyse with MediaInfo.
+     */
+    virtual void analyseMediaInfo(const char* inputFilepath);
+
+    /**
+     * @brief Encrypt the file or a portion of it
+     *
+     * Call this function once with the file to be uploaded.  It uses mediainfo to extract information that will
+     * help the webclient show or play the file in various browsers.  The information is stored in this object
+     * until the whole operation completes.   The encrypted data is stored in a new file.
+     *
+     * In order to save space on mobile devices, this function can be called in such a way that the last portion
+     * of the file is encrypted (to a new file), and then that last portion of the file is removed by file truncation.
+     * That operation can be repeated until the file is completely encrypted, and only the encrypted version remains,
+     * and takes up the same amount of space on the device.   The size of the portions must first be calculated by using
+     * the 'adjustsizeonly' parameter, and iterating from the start of the file, specifying the approximate sizes of the portions.
+     *
+     * Encryption is done by reading small pieces of the file, encrypting them, and outputting to the new file,
+     * so that RAM usage is not excessive.
+     *
+     * @param inputFilepath The file to encrypt a portion of (and the one that is ultimately being uploaded).
+     * @param startPos The index of the first byte of the file to encrypt
+     * @param length The number of bytes of the file to encrypt.  The function will round this value up by up to 1MB to fit the
+     *        MEGA internal chunking algorithm.  The number of bytes acutally encrypted and stored in the new file is the updated number.
+     * @param outputFilepath The name of the new file to create, and store the encrypted data in.
+     * @param urlSuffix The function will update the string passed in.  The content of the string must be appended to the URL
+     *        when this portion is uploaded.
+     * @param adjustsizeonly If this is set true, then encryption is not performed, and only the length parameter is adjusted.
+     *        This feature is to enable precalculating the exact sizes of the file portions for upload.
+     */
+    virtual bool encryptFile(const char* inputFilepath, int64_t startPos, unsigned int* length, const char* outputFilepath, std::string* urlSuffix, bool adjustsizeonly);
```

Comment:
The urlSuffix is an output parameter, and the pattern in the SDK seems to be to use std::string* in those cases?  Whereas the `const char*` parameters are input parameters.

Reply:
Correct me if I'm wrong, but the only method in `MegaApi` that uses a pointer to `std::string` is a converter UTF8<-->UTF16 used by WP8. It's only doable in Windows, it will not work in Android.
If we want to provide this feature only for iOS, it should be fine. If you want to make it compatible with other languages (specially those not C-friendly like Java or others), I recommend to change it. Otherwise, just mark this conversation as _Resolved_ :)

Reply:
Good point, I had not realised that about the java bindings.   I've made the update, those return a string now instead, with an empty string indicating failure.  thanks!

Reply:
Thanks. I would have preferred the old-fashion `char *`/`const char *` to easy the bindings (IIRC, those types don't need anything special, while `string` will need specific declaration for memory management), but it's fine. We may need to change in a future though